File: app/src/main/java/com/luphihung/mhike/model/Hike.java
package com.luphihung.mhike.model;

import java.io.Serializable;

/**
 * Represents a planned hike. Instances are passed between activities
 * (hence Serializable) and persisted in the local SQLite database.
 */
public class Hike implements Serializable {

    /** Value used for hikes that have not been saved to the database yet. */
    public static final long UNSAVED_ID = -1;

    private long id = UNSAVED_ID;
    private String name;
    private String location;
    /** Stored in ISO format (yyyy-MM-dd) so dates sort and compare correctly. */
    private String date;
    private boolean parkingAvailable;
    private double lengthKm;
    private String difficulty;
    private String description;
    // Custom fields required by the specification ("fields of your own invention")
    private double estimatedDurationHours;
    private String terrainType;

    public Hike() {
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public boolean isParkingAvailable() {
        return parkingAvailable;
    }

    public void setParkingAvailable(boolean parkingAvailable) {
        this.parkingAvailable = parkingAvailable;
    }

    public double getLengthKm() {
        return lengthKm;
    }

    public void setLengthKm(double lengthKm) {
        this.lengthKm = lengthKm;
    }

    public String getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(String difficulty) {
        this.difficulty = difficulty;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getEstimatedDurationHours() {
        return estimatedDurationHours;
    }

    public void setEstimatedDurationHours(double estimatedDurationHours) {
        this.estimatedDurationHours = estimatedDurationHours;
    }

    public String getTerrainType() {
        return terrainType;
    }

    public void setTerrainType(String terrainType) {
        this.terrainType = terrainType;
    }
}

File: MHike.Maui/Models/Hike.cs
using SQLite;

namespace MHike.Maui.Models;

/// <summary>
/// A planned hike stored in the local SQLite database.
/// Mirrors the schema used by the native Android version of M-Hike.
/// </summary>
[Table("hikes")]
public class Hike
{
    [PrimaryKey, AutoIncrement]
    [Column("id")]
    public int Id { get; set; }

    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("location")]
    public string Location { get; set; } = string.Empty;

    [Column("hike_date")]
    public DateTime Date { get; set; } = DateTime.Today;

    [Column("parking_available")]
    public bool ParkingAvailable { get; set; }

    [Column("length_km")]
    public double LengthKm { get; set; }

    [Column("difficulty")]
    public string Difficulty { get; set; } = string.Empty;

    [Column("description")]
    public string Description { get; set; } = string.Empty;

    // Custom fields required by the specification
    [Column("duration_hours")]
    public double EstimatedDurationHours { get; set; }

    [Column("terrain_type")]
    public string TerrainType { get; set; } = string.Empty;

    /// <summary>Summary line shown underneath the hike name in the list.</summary>
    [Ignore]
    public string Summary => $"{Location}  ·  {Date:ddd, dd MMM yyyy}  ·  {LengthKm} km";
}
